internal/gateway/connector/atomic_transform.go

```
@@ -0,0 +1,425 @@
+	atomicTransformBeforeCommitTestHooks.RLock()
+	hook := atomicTransformBeforeCommitTestHooks.byPath[key]
+	atomicTransformBeforeCommitTestHooks.RUnlock()
+	if hook != nil {
+		hook(attempt)
+	}
+}
+
+// Keep errors imported through errors.Is in callers without exposing the
+// snapshot implementation. This helper also normalizes Windows path-not-found
+// variants wrapped by os.PathError.
+func atomicTransformIsNotExist(err error) bool {
```

golangci-lint (2.12.2)

[error] func atomicTransformIsNotExist is unused

(unused)